src/core.rs

```diff
@@ -1210,22 +1210,22 @@ impl CoseKeySetOrKey<'_> {
         }
     }
 
-    pub fn as_key_set(&self) -> Option<&Vec<CoseKey<'_>>> {
+    pub fn as_key_set(&self) -> Option<&Vec<CoseKey>> {
```

Comment:
Do you think we should return a slice here instead of a reference to the Vector?

```suggestion
    pub fn as_key_set(&self) -> Option<&[CoseKey]> {
```